invenio_rdm_records/resources/iiif.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""IIIF Resource."""
+# TODO move this somewhere else if needed
+
+from abc import ABC, abstractmethod
+
+import requests
+from flask import Response, request
+from flask_resources import resource_requestctx
+
+
+class IIIFProxy(ABC):
+    """IIIF Proxy interface.
+
+    The purpose of this class is to provide a consistent way of proxying
+    requests to an IIIF server, regardless of the specific implementation
+    details of the server proxy. It defines a set of methods that should be
+    implemented by the server proxy, such as `proxy_request` and
+    `handle_url_rewrite`.
+
+    To use this class, you should create a subclass that implements the
+    required methods. The `IIIFProxy` class provides some default
+    implementations for certain methods, such as `should_proxy`. These default implementations can be
+    overridden in the subclass if needed.
+
+    Example usage:
+
+    .. code-block::python
+
+        class MyIIIFProxy(IIIFProxy):
+        def proxy_request(self):
+            # Implementation specific to the IIIF server proxy
+
+        def handle_url_rewrite(self):
+            # Implementation specific to the IIIF server proxy
+
+        # Instantiate the proxy
+        proxy = MyIIIFProxy()
+
+        # Proxy the current request if it should be proxied
+        proxy()
+    """
+
+    @property
+    def proxied_routes(self):
+        """List of routes that should be proxied."""
+        return []
+
+    @property
+    def server_url(self):
+        """IIIF server URL."""
+        raise NotImplementedError("IIIF server must be set.")
+
+    @abstractmethod
+    def proxy_request(self):
+        """Proxy the current request to IIIF server."""
+
+    @abstractmethod
+    def handle_url_rewrite(self):
+        """Handle URL rewrite."""
+
+    def should_proxy(self):
+        """Check if the curent request should be proxied."""
+        return False
+
+    def __call__(self):
+        """Proxy request to IIIF server if the endpoint is configured."""
+        if self.should_proxy():
+            return self.proxy_request()
+        return None
+
+
+class IIPServerProxy(IIIFProxy):
```

Comment:
For this implementation to be useful for broader use, we need to make `server_url` and `handle_url_rewrite` configurabl